To: vim_dev@googlegroups.com
Subject: Patch 7.3.122
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.122
Problem:    Having auto/config.mk in the repository causes problems.
Solution:   Remove auto/config.mk from the distribution.  In the toplevel
	    Makefile copy it from the "dist" file.
Files:	    Makefile, src/Makefile, src/auto/config.mk


*** ../vim-7.3.121/Makefile	2010-08-15 21:57:20.000000000 +0200
--- Makefile	2011-02-12 14:28:09.000000000 +0100
***************
*** 23,28 ****
--- 23,31 ----
  # has run can result in compiling with $(CC) empty.
  
  first:
+ 	@if test ! -f src/auto/config.mk; then \
+ 		cp src/config.mk.dist src/auto/config.mk; \
+ 	fi
  	@echo "Starting make in the src directory."
  	@echo "If there are problems, cd to the src directory and run make there"
  	cd src && $(MAKE) $@
***************
*** 30,35 ****
--- 33,41 ----
  # Some make programs use the last target for the $@ default; put the other
  # targets separately to always let $@ expand to "first" by default.
  all install uninstall tools config configure reconfig proto depend lint tags types test testclean clean distclean:
+ 	@if test ! -f src/auto/config.mk; then \
+ 		cp src/config.mk.dist src/auto/config.mk; \
+ 	fi
  	@echo "Starting make in the src directory."
  	@echo "If there are problems, cd to the src directory and run make there"
  	cd src && $(MAKE) $@
*** ../vim-7.3.121/src/Makefile	2010-11-03 22:32:18.000000000 +0100
--- src/Makefile	2011-02-15 15:21:37.000000000 +0100
***************
*** 283,294 ****
  
  ######################## auto/config.mk ######################## {{{1
  # At this position auto/config.mk is included. When starting from the
! # distribution it is almost empty. After running auto/configure it contains
! # settings that have been discovered for your system. Settings below this
! # include override settings in auto/config.mk!
! 
! # Note: if auto/config.mk is lost somehow (e.g., because configure was
! # interrupted), create an empty auto/config.mk file and do "make config".
  
  # (X) How to include auto/config.mk depends on the version of "make" you have,
  #     if the current choice doesn't work, try the other one.
--- 283,295 ----
  
  ######################## auto/config.mk ######################## {{{1
  # At this position auto/config.mk is included. When starting from the
! # toplevel Makefile it is almost empty. After running auto/configure it
! # contains settings that have been discovered for your system. Settings below
! # this include override settings in auto/config.mk!
! 
! # Note: If make fails because auto/config.mk does not exist (it is not
! # included in the repository), do:
! #    cp config.mk.dist auto/config.mk
  
  # (X) How to include auto/config.mk depends on the version of "make" you have,
  #     if the current choice doesn't work, try the other one.
*** ../vim-7.3.121/src/auto/config.mk	2010-08-16 21:59:00.000000000 +0200
--- src/auto/config.mk	1970-01-01 01:00:00.000000000 +0100
***************
*** 1,5 ****
- the first targets to make vim are: scratch config myself
- srcdir = .
- VIMNAME = vim
- EXNAME = ex
- VIEWNAME = view
--- 0 ----
*** ../vim-7.3.121/src/version.c	2011-02-15 14:24:42.000000000 +0100
--- src/version.c	2011-02-15 15:25:07.000000000 +0100
***************
*** 716,717 ****
--- 716,719 ----
  {   /* Add new patch number below this line */
+ /**/
+     122,
  /**/

-- 
hundred-and-one symptoms of being an internet addict:
267. You get an extra phone line so you can get phone calls.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
